src/components/apps/admin/referenceGenomes/TableView.js

```diff
@@ -0,0 +1,193 @@
+
+import { build } from "@cyverse-de/ui-lib";
+
+import Checkbox from "@material-ui/core/Checkbox";
+import MaUTable from "@material-ui/core/Table";
+import Paper from "@material-ui/core/Paper";
+
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableSortLabel from "@material-ui/core/TableSortLabel";
+
+const IndeterminateCheckbox = React.forwardRef(
+    ({ indeterminate, ...rest }, ref) => {
+        const defaultRef = React.useRef();
+        const resolvedRef = ref || defaultRef;
+
+        React.useEffect(() => {
+            resolvedRef.current.indeterminate = indeterminate;
+        }, [resolvedRef, indeterminate]);
+
+        return (
+            <>
+                <Checkbox ref={resolvedRef} {...rest} />
+            </>
+        );
+    }
+);
+
+const EnhancedTable = ({
+    baseId,
+    columns,
+    data,
+    onAddClicked,
+    onDeletedClicked,
+}) => {
+    const {
+        getTableProps,
+        headerGroups,
+        prepareRow,
+        rows,
+        preGlobalFilteredRows,
+        setGlobalFilter,
+        //selectedFlatRows,
+        state: { /*selectedRowIds,*/ globalFilter },
+        toggleRowSelected,
+    } = useTable(
+        {
+            columns,
+            data,
+            initialState: {
+                sortBy: [
+                    {
+                        id: refGenomeConstants.keys.NAME,
+                        desc: false,
+                    },
+                ],
+            },
+        },
+        useGlobalFilter,
+        useSortBy,
+        useRowSelect,
+        (hooks) => {
+            hooks.allColumns.push((columns) => [
+                // Let's make a column for selection
+                {
+                    id: "selection",
+                    // The header can use the table's getToggleAllRowsSelectedProps method
+                    // to render a checkbox
+                    Header: ({ getToggleAllRowsSelectedProps }) => (
+                        <div>
+                            <IndeterminateCheckbox
+                                {...getToggleAllRowsSelectedProps()}
+                            />
+                        </div>
+                    ),
+                    // The cell can use the individual row's getToggleRowSelectedProps method
+                    // to the render a checkbox
+                    Cell: ({ row }) => (
+                        <div>
+                            <IndeterminateCheckbox
+                                {...row.getToggleRowSelectedProps()}
+                            />
+                        </div>
+                    ),
+                },
+                ...columns,
+            ]);
+        }
+    );
+
+    const tableId = build(baseId, ids.TABLE_VIEW);
+    /*  useEffect(() => {
```

Comment:
More commented out code that could be nuked, possibly.

Reply:
makes sense. Mostly left for folks who are new to `react-table` stuff. I can nuke it before merge.